### ISSUE TITLE
Exult Studio: Files/Groups list need to be scrollable

### ISSUE THREAD

**Issue by DominusExult** [exult/exult#249]
Exult Studio may become hard to use because the files/groups list can't be made smaller than the list needs to be when every file type is uncollapsed.
Example: 
- Open a Serpent Isle game in Exult Studio
- now try to resize the height of the Exult Studio window
- It will not get smaller than the last file in the list (unless you collapse Palette Files)

This is a problem on smaller laptop screens (1366x768) or when you have a lot of extra files in Shape Files, Palette Files, ...
The person who reported it was desperate because he couldn't resize the window and thus not use the buttons on the bottom.
Only after trying a lot of things  did we realize that this was the culprit.

@Dragon-Baroque could you take a look? I guess you are the most familiar at the moment.

![image](https://user-images.githubusercontent.com/3313896/190685459-651080f9-c642-4271-939a-a6f409cb941f.png)

**Comment by Dragon-Baroque** [comment]
@DominusExult, there is a very simple change that I would like you to perform and tell me whether it suits you.
Change with a regular text editor in the `share/exult/exult_studio.glade` the Vertical Scroll Policy of the `scrolledwindow2` from `never` to `always` :
```
...
              <object class="GtkScrolledWindow" id="scrolledwindow2">
                <property name="visible">True</property>
                <property name="can-focus">False</property>
                <property name="halign">fill</property>
                <property name="valign">fill</property>
                <property name="margin-start">4</property>
                <property name="margin-end">4</property>
                <property name="margin-top">2</property>
                <property name="margin-bottom">2</property>
                <property name="hexpand">False</property>
                <property name="vexpand">False</property>
                <property name="hscrollbar-policy">never</property>
                <property name="vscrollbar-policy">never</property>
...
```
You should alter the last line of that excerpt to
```
                <property name="vscrollbar-policy">always</property>·
```
Then run Exult Studio again.
This, for me, creates a vertical scroll bar on the file / group treeview, visible when I hover the mouse over its expected place at the right of the treeview.
But it makes also the treeview _very, very small_ vertically, I mean only five rows or so are visible.
Does it suit you ?
If so I can push a commit to patch the glade file

**Comment by DominusExult** [comment]
Works fine, also works with setting it to automatic. I also see the smallness of the vertical treeview, can there be a default expansion? Or maybe a default for the whole Studio windows (as it is saved on closing in exult.cfg)?
Another odditty in regards to horizonal view, it always seems to start with a reduced view that you can manually adjust, but when you select a file it automatically expands to a set horizontal view that cannot be adjusted anymore.

**Comment by Dragon-Baroque** [comment]
> Works fine, also works with setting it to automatic.

True. `automatic` should be better than `always`, because `always` presents a scrollbar even when it is unneeded ( and then useless ).

> I also see the smallness of the vertical treeview, can there be a default expansion? Or maybe a default for the whole Studio windows (as it is saved on closing in exult.cfg)?

All my tests have been performed after a removal of `.exult.cfg`. To be safe.
The GtkTreeView is fully embedded in a GtkScrolledWindow, because the GtkScrolledWindow can carry scrollbars, and the GtkTreeView cannot.
I did not search much, but I have not been able to act on the GtkTreeView properties. On the other hand, we can add a min vertical size to the GtkScrolledWindow, in pixels, to the properties of the `scrolledwindow2`, for example :
```
  <property name="min-content-height">360</property>
```
With `360`, the base `Forge of Virtue` fits, no scrollbar. With the `Keyring` mod, its does no longer fit and provides the scrollbar. With `Silver Seed`, with or without mods, it does not fit either.
> Another oddity in regards to horizontal view, it always seems to start with a reduced view that you can manually adjust, but when you select a file it automatically expands to a set horizontal view that cannot be adjusted anymore.

Something interesting : with my Exult Studio experiment under GTK 4, the GtkTreeView starts with the large width, and the width _reduces_ when I select any other file than `SHAPES.VGA`. When I alternate between `SHAPES.VGA` and any other file, the width enlarges and reduces accordingly. Further study needed...

**Comment by DominusExult** [comment]
` <property name="min-content-height">360</property>` it should be then, I think that's a good preset and that is only until you adjust it and the new default is in exult.cfg. Thanks for figuring a good way out :)

Sorry for piling on in this topic, but I wonder if the persistent scroll bar of the actual browser should be done away and be replaced by an automatic one. I found that you can only scroll down (with trackpad or mouse scroll wheel) when you are hovering on that scroll bar. 

Interesting that the GTK4 view scrambles other things up again...

**Comment by Dragon-Baroque** [comment]
Proposal : remove property `vscrollbar-policy` because `automatic` is the default, replace by property `min-content-height` set to `360`. I shall make a Pull Request very shortly along these lines.

> I wonder if the persistent scroll bar of the actual browser should be done away and be replaced by an automatic one.

Mmh. The scrolling logic of the Shape Browser GtkDrawingArea is quite different from, say, the File List GtkTreeView.
The File List is produced by the Game Chooser as a fixed size image, displayed whole width, with a fixed width ( subject to the unexpected width changes we have noticed ), and vertically scrollable at pixel precision.
The Shape Browser has no fixed width, it reacts to width changes by displaying more or less shapes per row. In addition, the vertical scroll ensures that full shapes are displayed.

1. I can try to add the missing mouse and keyboard events to the Shape Browser GtkDrawingArea to provide scrolling. It already handles the mouse clicks ( select and open the Shape Editor ), and some keyboard events ( Insert and Delete keys ). I think this is more compliant with the GTK style : attach some event controllers to the widgets. I could then even remove the scrollbar(s).
2. I can try to embed the Shape Browser into a GtkScrolledWindow with only a vertical scrolling but I should somehow allow its width to change and order a redraw of the GtkDrawingArea when this occurs, and I should ensure that the vertical scrolling steps to Shape boundaries. Tricky, I think.

**Comment by Dragon-Baroque** [comment]
> Another oddity in regards to horizontal view, it always seems to start with a reduced view that you can manually adjust, but when you select a file it automatically expands to a set horizontal view that cannot be adjusted anymore.

Ah. I have understood why the width changes with the `SHAPES.VGA`. It is because this file has builtin groups, and the width of the Combo Box that contains the builtin group names with such names as `Objects with Gump` along with the other widgets of the Group page requires a larger width of the Group notebook page.

If you agree, I can increase the `width-request` of the `frame120` from `200` to `280` or so, this frame is the first child of the left pane of the main window, it immediately contains the File / Group notebook. With that you encompass the `SHAPES.VGA` builtin group names from the start.

**Comment by DominusExult** [comment]
> If you agree, I can increase the width-request of the frame120 from 200 to 280 or so, this frame is the first child of the left pane of the main window, it immediately contains the File / Group notebook. With that you encompass the SHAPES.VGA builtin group names from the start.

Thanks for analyzing this!


> I can try to add the missing mouse and keyboard events to the Shape Browser GtkDrawingArea to provide scrolling. It already handles the mouse clicks ( select and open the Shape Editor ), and some keyboard events ( Insert and Delete keys ). I think this is more compliant with the GTK style : attach some event controllers to the widgets. I could then even remove the scrollbar(s).
> I can try to embed the Shape Browser into a GtkScrolledWindow with only a vertical scrolling but I should somehow allow its width to change and order a redraw of the GtkDrawingArea when this occurs, and I should ensure that the vertical scrolling steps to Shape boundaries. Tricky, I think.

This all sounds very time consuming. I thought it would be easy piecey like the file list issue.

**Comment by Dragon-Baroque** [comment]
1. About the File List changing width, perhaps the most simple way is to make _visible_ the Groups page at the start, as I have now made sure that the display of this page increases the File List width.
If you want to test, this is, in the `share/exult/exult_studio.glade`, from its `exult/exult` state, just add the second line after the first :
```
  <object class="GtkFrame" id="groups_frame">
    <property name="visible">True</property>
```
2. Scrolling from inside the Shapes GtkDrawingArea. I have made a wrong statement, namely that the vertical scrollbar at its right scrolls by entire rows of shapes. Not so. It performs a smooth scrolling and can display ( vertically ) cut shapes. So I gave myself a harder problem by willing to scroll by rows. Also I made the request more complex than your initial question, namely to react also to keyboard events Up ( 1 row up ), Page Up ( 4 rows up ), Down (1 row down ), Page Down ( 4 rows down ). I am attaching a patch [ to mapedit/shapelst.h and cc ] that performs both. You will notice that the largest piece of code in the patch is the one row shift up or down. The Mouse wheel event handler is trivial, and the Keyboard event handler adds just four keys.

[studio_scroll.zip](https://github.com/exult/exult/files/9598918/studio_scroll.zip)

One patched and recompiled, open Studio, choose SHAPES.VGA, the mouse wheel should work from inside the Shapes GtkDrawingArea.
To get the Up / Down / Page Up / Page Down keys to work, you probably need to switch the keyboard from the File List to the Shapes display, I use `Tab` for that.

And it would be even easier to simulate a smooth scrolling, if you prefer...

**Comment by Dragon-Baroque** [comment]
I forgot above : The patch applies to `shapelst`. Thus it works for all files of the file list, except for the Groups ( code to be replicated ), the Chunks, Combos or NPCs ( code to be replicated ), or Palette ( irrelevant, no scrolling ).

**Comment by Dragon-Baroque** [comment]
Here it is. New patch to act in the shape browser with the mouse wheel like with the vertical scrollbar.
[studio_scroll_2.zip](https://github.com/exult/exult/files/9601846/studio_scroll_2.zip)
Same as the previous one : apply patch to Exult as `master`.
Please report whether you prefer as that, whether you get stalls, and whether you find the ( Page ) Up / Down keys any useful ( if not, this would simplify the patch way more and make it very easy to replicate into Chunks, Combos, NPCs and Groups ).

**Comment by DominusExult** [comment]
> Bigger in width than when it reached its max size on SHAPES.VGA and when you select a file it automatically expands to a set horizontal view that cannot be adjusted anymore ?

Yes, it is too big in width, and from my messing with the the glade file and what you mentioned before, I think it is because it expands to the size of the "Groups" lower part elements (text entry box, New, Delete). Probably not much one can do about that as you need a sensible default for the text entry box. (if that is the cause of this)

> Here it is. New patch to act in the shape browser with the mouse wheel like with the vertical scrollbar.

I cannot test this today and the soonest tomorrow evening :(
However I do find the (page) up/down useful but OTOH I did not miss these before you gave me the opportunity to use. I'm not sure whether I'd really use them

**Comment by Dragon-Baroque** [comment]
> I think it is because it expands to the size of the "Groups" lower part elements (text entry box, New, Delete).

Exactly. A GTK+ 3 GtkEntry cannot be narrower that **150 pixels**. Add to it the New and Delete buttons and you reach the width of the File List. We could reduce the final width if we changed the layout of the bottom, say, the two buttons _below_ the GtkEntry,
By the way, this is the same on the right pane, as the Find, Locate F: and Locate Q: GtkEntrys are responsible for the width of the Shape browser.
My experiments with GTK 4 produce quite narrower File List and Shape Browser frames, because the same GtkEntrys are much more narrow.

**Comment by DominusExult** [comment]
> Here it is. New patch to act in the shape browser with the mouse wheel like with the vertical scrollbar.
> [studio_scroll_2.zip](https://github.com/exult/exult/files/9601846/studio_scroll_2.zip)
> Same as the previous one : apply patch to Exult as `master`.
> Please report whether you prefer as that, whether you get stalls, and whether you find the ( Page ) Up / Down keys any useful ( if not, this would simplify the patch way more and make it very easy to replicate into Chunks, Combos, NPCs and Groups ).

This works way better, no stalls, in fact it is a bit too fast compared to how the scroll bar scrolls.
(page) up/down might be useful but as I wrote before, I'm not sure I'd actually use it.

**Comment by Dragon-Baroque** [comment]
> It is a bit too fast compared to how the scroll bar scrolls.

This is unexpected. I tried to ensure that the vertical shift of the Shape browser remained the same at one click of the mouse wheel in the browser page and in the scroller.

_However, I have noticed a rounding error, so that I am posting a new patch._
[studio_scroll_3.zip](https://github.com/exult/exult/files/9617686/studio_scroll_3.zip)

If you feel that you still get a difference in speed, I would like you to quantify that statement :

1. You should understand that my patch creates a wheel event handler to the browser that computes and applies a displacement on the scroller. Then the scroller posts an event to Studio handled by the code that shifts the browser, I have not changed that later code.
2. Studio displays messages onto `cout` during scroll events, provided you launch Studio from the command like. Like this :

```
Scrolled to 11.5195
Scrolled to 23.039
...
Scrolled to 334.065
Scrolled to 345.584
```
The messages above come from my mouse drag actions on the scroller. About 11.5 pixels per event.
```
Scrolled to 407.781
Scrolled to 469.977
Scrolled to 532.174
Scrolled to 594.37
Scrolled to 656.567
Scrolled to 718.763
```
The messages above come from my mouse wheel actions on the scroller. About 62.2 pixels per event, with one event per mouse wheel click.
```
Scroll Event -1.5, value 718.763 -> 656.567 in 0 .. 2928
Scrolled to 656.567
Scroll Event -1.5, value 656.567 -> 594.37 in 0 .. 2928
Scrolled to 594.37
Scroll Event -1.5, value 594.37 -> 532.174 in 0 .. 2928
Scrolled to 532.174
Scroll Event -1.5, value 532.174 -> 469.977 in 0 .. 2928
Scrolled to 469.977
Scroll Event -1.5, value 469.977 -> 407.781 in 0 .. 2928
Scrolled to 407.781
Scroll Event -1.5, value 407.781 -> 345.584 in 0 .. 2928
Scrolled to 345.584
Scroll Event 1.5, value 345.584 -> 407.781 in 0 .. 2928
Scrolled to 407.781
Scroll Event 1.5, value 407.781 -> 469.977 in 0 .. 2928
Scrolled to 469.977
Scroll Event 1.5, value 469.977 -> 532.174 in 0 .. 2928
Scrolled to 532.174
Scroll Event 1.5, value 532.174 -> 594.37 in 0 .. 2928
Scrolled to 594.37
Scroll Event 1.5, value 594.37 -> 656.567 in 0 .. 2928
Scrolled to 656.567
Scroll Event 1.5, value 656.567 -> 718.763 in 0 .. 2928
Scrolled to 718.763
```
The messages above come from my mouse wheel actions **in the browser**. These actions produce two messages for each click :

1. One more verbose from the browser scroll event handler, where I report plus or minus 1.5 mouse click units ( instead of 1.0 I would have expected, please don't ask me why, this may be due how my mouse is configured ), and the current scroller value and the value I intend to set it too, finally the whole scroller range.
2. The normal scroll event handle, caused by the scroller value change, to the value I intended to set the scroller to.

Each pair of messages reports an increase or a decrease of 62.2 pixels. Moreover you notice that I end up with the same pixel shifts as with the mouse wheel clicks **in the scroller**.
```
Scrolled to 780.959
Scrolled to 843.156
Scroll Event -1.5, value 843.156 -> 780.959 in 0 .. 2928
Scrolled to 780.959
Scroll Event -1.5, value 780.959 -> 718.763 in 0 .. 2928
Scrolled to 718.763
Scroll Event -1.5, value 718.763 -> 656.567 in 0 .. 2928
Scrolled to 656.567
Scroll Event 1.5, value 656.567 -> 718.763 in 0 .. 2928
Scrolled to 718.763
Scroll Event 1.5, value 718.763 -> 780.959 in 0 .. 2928
Scrolled to 780.959
Scroll Event 1.5, value 780.959 -> 843.156 in 0 .. 2928
Scrolled to 843.156
Scroll Event 1.5, value 843.156 -> 905.352 in 0 .. 2928
Scrolled to 905.352
Scroll Event 1.5, value 905.352 -> 967.549 in 0 .. 2928
Scrolled to 967.549
Scroll Event 1.5, value 967.549 -> 1029.75 in 0 .. 2928
Scrolled to 1029.75
Scrolled to 967.549
Scrolled to 905.352
Scrolled to 843.156
```
Various other tests either with the mouse wheel in the scroller ( only one short message ) or in the browser ( one long message and one short message ). Again, I pass through the same pixel shifts.

Would you please perform the same tests ?  And does this occur on Windows or macOS or iOS ?

**Comment by DominusExult** [comment]
These are the tests, first with a mouse + wheel, then with the trackpad scroll gesture. This is all on macOS (no gtk on iOS).
Always started from position 0.
As you can see the scrolling in the browser quickly jumps to the end. Mouse wheel with smooth scrolling disabled the quickest (three clicks).
```
Scrolled to 1.30879
Scrolled to 2.61759
Scrolled to 3.92638
Scrolled to 5.23517
Scrolled to 7.85276
Scrolled to 9.16155
Scrolled to 10.4703
Scrolled to 11.7791
Scrolled to 13.0879
Scrolled to 14.3967
Scrolled to 15.7055
Scrolled to 17.0143
Scrolled to 18.3231
Scrolled to 19.6319
Scrolled to 20.9407
Scrolled to 22.2495
Scrolled to 23.5583

```
The messages above come from my mouse drag actions on the scroller. 


```
Scrolled to 13
Scrolled to 26
Scrolled to 39
Scrolled to 52
Scrolled to 65
Scrolled to 78
Scrolled to 91
Scrolled to 104
Scrolled to 117
Scrolled to 130
Scrolled to 143
Scrolled to 156
Scrolled to 169
Scrolled to 182

```
The messages above come from my mouse wheel actions on the scroller.

```
Scrolled to 1
Scrolled to 3
Scrolled to 4
Scrolled to 6
Scrolled to 9
Scrolled to 11
Scrolled to 12
Scrolled to 14
Scrolled to 16
Scrolled to 19
Scrolled to 24
Scrolled to 25
Scrolled to 30
Scrolled to 39
Scrolled to 41
Scrolled to 47
Scrolled to 59
Scrolled to 61
Scrolled to 64
Scrolled to 65
Scrolled to 67
Scrolled to 72
Scrolled to 74
```
The messages above come from my mouse wheel actions on the scroller with smooth scrolling enabled

```
Scroll Event 13, value 0 -> 955.367 in 0 .. 2663
Scrolled to 955.367
Scroll Event 13, value 955.367 -> 1910.73 in 0 .. 2663
Scrolled to 1910.73
Scroll Event 13, value 1910.73 -> 2866.1 in 0 .. 2663
Scrolled to 2033
```
The messages above come from my mouse wheel actions in the browser.

```
Scroll Event 2, value 0 -> 146.98 in 0 .. 2663
Scrolled to 146.98
Scroll Event 7, value 146.98 -> 661.408 in 0 .. 2663
Scrolled to 661.408
Scroll Event 1, value 661.408 -> 734.898 in 0 .. 2663
Scrolled to 734.898
Scroll Event 2, value 734.898 -> 881.878 in 0 .. 2663
Scrolled to 881.878
Scroll Event 1, value 881.878 -> 955.367 in 0 .. 2663
Scrolled to 955.367
Scroll Event 2, value 955.367 -> 1102.35 in 0 .. 2663
Scrolled to 1102.35
Scroll Event 2, value 1102.35 -> 1249.33 in 0 .. 2663
Scrolled to 1249.33
Scroll Event 1, value 1249.33 -> 1322.82 in 0 .. 2663
Scrolled to 1322.82
Scroll Event 4, value 1322.82 -> 1616.78 in 0 .. 2663
Scrolled to 1616.78
Scroll Event 1, value 1616.78 -> 1690.27 in 0 .. 2663
Scrolled to 1690.27
Scroll Event 2, value 1690.27 -> 1837.24 in 0 .. 2663
Scrolled to 1837.24
Scroll Event 1, value 1837.24 -> 1910.73 in 0 .. 2663
Scrolled to 1910.73
Scroll Event 5, value 1910.73 -> 2278.18 in 0 .. 2663
Scrolled to 2033
```
The messages above come from my mouse wheel actions in the browser with smooth scrolling enabled.

```
Scrolled to 4.15746
Scrolled to 5.46626
Scrolled to 6.77505
Scrolled to 8.08384
Scrolled to 9.39264
Scrolled to 10.7014
Scrolled to 12.0102
Scrolled to 13.319
Scrolled to 14.6278
Scrolled to 15.9366
Scrolled to 17.2454
Scrolled to 18.5542
Scrolled to 19.863
Scrolled to 21.1718
Scrolled to 22.4806
Scrolled to 23.7894
Scrolled to 25.0982

```
The messages above come from my touchpad drag actions on the scroller. 

```
Scrolled to 1
Scrolled to 9
Scrolled to 22
Scrolled to 23
Scrolled to 26
Scrolled to 30
Scrolled to 34
Scrolled to 38
Scrolled to 42
Scrolled to 46
Scrolled to 50
Scrolled to 54
Scrolled to 58
Scrolled to 62
Scrolled to 66
Scrolled to 70
Scrolled to 77
```
The messages above come from my touchpad scroll actions on the scroller.

```
Scroll Event 1, value 0 -> 73.4898 in 0 .. 2663
Scrolled to 73.4898
Scroll Event 8, value 73.4898 -> 661.408 in 0 .. 2663
Scrolled to 661.408
Scroll Event 3, value 661.408 -> 881.878 in 0 .. 2663
Scrolled to 881.878
Scroll Event 2, value 881.878 -> 1028.86 in 0 .. 2663
Scrolled to 1028.86
Scroll Event 4, value 1028.86 -> 1322.82 in 0 .. 2663
Scrolled to 1322.82
Scroll Event 4, value 1322.82 -> 1616.78 in 0 .. 2663
Scrolled to 1616.78
Scroll Event 3, value 1616.78 -> 1837.24 in 0 .. 2663
Scrolled to 1837.24
Scroll Event 1, value 1837.24 -> 1910.73 in 0 .. 2663
Scrolled to 1910.73
Scroll Event 1, value 1910.73 -> 1984.22 in 0 .. 2663
Scrolled to 1984.22
Scroll Event 1, value 1984.22 -> 2057.71 in 0 .. 2663
Scrolled to 2033
```
The messages above come from my touchpad scroll actions in the browser.

**Comment by Dragon-Baroque** [comment]
Thank you, @DominusExult.
I see the difference. Whereas in my Linux setup, mouse wheel clicks units have to be converted into pixels ( multiply by a few tens ), it looks like in macOS, mouse / touchpad scroll events are already in pixels.

To remove the spurious conversion :
After you apply my studio_scroll_3.diff, there is this in `mapedit/shapelst.cc`, line 1892+, function `Shape_chooser::scroll_some_rows`
```
void Shape_chooser::scroll_some_rows(
    gdouble count
) {
    GtkAdjustment *adj = gtk_range_get_adjustment(
        GTK_RANGE(vscroll));
    const gdouble adj_pgsize = gtk_adjustment_get_page_size(adj);
    const gdouble adj_lower  = gtk_adjustment_get_lower(adj);
    const gdouble adj_upper  = gtk_adjustment_get_upper(adj);
    const gdouble adj_value  = gtk_adjustment_get_value(adj);
    const gdouble new_value  = (count * pow(adj_pgsize, 2.0/3.0))
        + adj_value;
...
```
To remove my conversion ( the multiply by `pagesize ** 2/3` ), please alter the last line of the listing above to
```
    const gdouble new_value  = count + adj_value;
```
It should work better. Now I have to dig further to understand the difference Linux / macOS.
The `with smooth scrolling enabled`, is it in macOS settings ?

**Comment by DominusExult** [comment]
I'll give it a try. The smooth scrolling is an option of the Logitech mouse. Disabled the wheel has noticeable "clicks"/physical resistance. Enabled it spins freely until stopped.

**Comment by marzojr** [comment]
The difference could be due to highdpi stuff.

**Comment by DominusExult** [comment]
With this change both browser and scroll bar scrolling work the same way

**Comment by Dragon-Baroque** [comment]
Thank you, @DominusExult.

I think at this point that I am ready to work on a patch and a Pull Request for :

1. A correct working of the mouse wheel on all platforms, with the help of your `#if defined`,
2. Without the support of the arrow and page keys, since I cannot any longer emulate a mouse wheel click on macOS, and since you do not feel this to be crucial,
3. Replicated to the Combos, the Chunks, the NPCs, and the Group shape windows,
4. ( For myself ) Made compliant with GTK 4.

This shall take a few days too...

**Comment by Dragon-Baroque** [comment]
Closed issue by PR #253

**Comment by DominusExult** [comment]
works, great! thanks a lot for doing that!

**Comment by Dragon-Baroque** [comment]
Forgive me @DominusExult, I may have closed the issue a little too quickly.
There remains a choice between 

1. the initial FileList treeview begin narrow at first and enlarging itself once you click on any file, because it then displays the Group tab and prepares the much wider Group page for display,
2. or the initial FileList treeview having from the start its wider size because the Group tab and page are ready for display at start. 

The advantage of the later is that there are no width expansion, but on the other hand, should you click on the Group tab at start with no selected file, Studio displays the `SHAPES.VGA` builtin groups, which may be surprising.

The difference, you remember, is in `share/exult/exult_studio.glade`, add a `visible` property to the `groups_frame`.

> <object class="GtkFrame" id="groups_frame">
    <property name="visible">True</property>

**Comment by DominusExult** [comment]
I vote for 2, now that I noticed the width jumping it annoys me a lot ;)

**Comment by Dragon-Baroque** [comment]
Implemented 2 by Pull Request #255, that I merged myself.

On the same vein, we could prevent the enlargement of the right panel by choosing a starting default browser. I would then suggest we default Studio to the `shapes.vga` at start, provided it exists, of course.
